Auto spell cast from Grimoire

pjie opened the pull request:
If you have 400 or more of each building, Spontaneous Edifice no longer has any valid buildings to create.

Can you possibly make the auto-spell function check for this, and sell chancemakers if there are no eligible buildings for Spontaneous Edifice?  

The optimal end-game strategy is actually to sit at 400 chancemakers with Earthshatterer aura, and then when mana is full, sell one chancemaker and get it back with Spontaneous Edifice.
https://www.reddit.com/r/CookieClicker/comments/797dmy/extreme_endgame_spontaneous_edifice_is_the/

briansmiley committed:
Fixing Godzamok cursor sell spam

briansmiley committed:
Update fc_main.js

briansmiley committed:
Added Phoenix09's sugar lump autoharvest

briansmiley committed:
Added Phoenix09's sugar lump autoharvesting

briansmiley committed:
Added Godzamok buy/sell change as menu option

Making the functionality where Auto-Godzamok avoids constantly buying/selling cursors as fast as possible (vs. waiting for a current Devastation buff to end before selling more) toggleable.

briansmiley committed:
Adding "Sane" option to Auto-Godzamok

If Auto-Godzamok is set to 1 (On), does rapidfire buy/sell. If set to 2 ('Sane'), waits for current Devastation buff to end before selling more, which feels less absurd.

briansmiley committed:
Merge pull request #1 from Chevron42/Experimenting

Godzamok Menu Toggle, Auto-SL Harvest

briansmiley committed:
Added "Show Achivement popups" toggle

briansmiley committed:
Added displaying achievement popups

briansmiley committed:
typo

briansmiley committed:
typo

briansmiley committed:
Fixing Show Achievements

briansmiley committed:
debugging show achievements

briansmiley committed:
still debugging

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Trying a different tack for toggling achievements

briansmiley committed:
Removed ShowAchievement functionality for now 

See comments on fcWin. Something weird is going on with the win function being called on the CPS achievements

briansmiley committed:
indicated that show achievements is broken early

briansmiley committed:
Adding auto spell casting

briansmiley committed:
AutoSpell

briansmiley committed:
Update cc_upgrade_prerequisites.js

briansmiley committed:
Actually implementing autocast

briansmiley committed:
Fixing autocast

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update cc_upgrade_prerequisites.js

briansmiley committed:
Added frenzy options to autocast

briansmiley committed:
Added frenzy options to autocast

briansmiley committed:
Fixing double autocast (increase interval)

briansmiley committed:
Removing my fcWin hack thing

briansmiley committed:
Removing "show achievements" toggle

briansmiley committed:
Frenzy multiplier autoCast criterion

briansmiley committed:
Adding Frenzy multiplier criterion to autocast

briansmiley committed:
Update cc_upgrade_prerequisites.js

briansmiley committed:
Update cc_upgrade_prerequisites.js

briansmiley committed:
Tweaking autocast frenzy stuff

briansmiley committed:
Adding Autocast frenzy prompt

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
AutoSpell now sells a Chancemaker if you have 400 of everything

briansmiley committed:
debugging

briansmiley committed:
Update fc_main.js

briansmiley committed:
Fixing AutoSpell

Fixing Autospell to sell 400th+ Chancemakers if necessary, and changed AutoBuy to hold necessary funds when Autocast Spontaneous edifice is selected (i.e. hold half the cost of the most expensive building under 400 in bank).

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Trying to fix SE stuff

briansmiley committed:
Update fc_main.js

briansmiley committed:
Prevent autobuy Chancemakers with SE autocast

Trying to implement removing Chancemakers from the purchase recommendation list when SE farming is turned on, to stay at 399 while still buying other buildings.

briansmiley committed:
typo in for loop

briansmiley committed:
Fixing autobuy/save for autocast

Trying to make the autoBuy function avoid going below the spontaneous edifice threshhold when autocast SE enabled (should save enough cookies for the most expensive building under 400)

briansmiley committed:
Fixing SE bank

Making the mostExpensive function properly check how much bank to aim for to enable SE (by defaulting to ~41 duodecillion when at 399 chancemakers)

briansmiley committed:
Blacklist Wizard Towers at 100 mana (if autoSpell)

517 Wizard towers give you 100 mana, and any more would slow down casting, so if autoSpell is enabled, Wizard towers cap at 517.

briansmiley committed:
Added a logEvent for SE Chancemaker sales

briansmiley committed:
Missing bracket

briansmiley commented:
Made some changes in addition to the one suggested by pjie

briansmiley committed:
Fixed syntax in SE autocast checking

Now should properly traverse Game.Objects to check for building prices/counts for SE casting conditions

briansmiley committed:
Missed one

briansmiley committed:
Adding Haggler's Charm AutoSpell

Since it's the cheapest spell, this adds the fastest option for getting 999 spells cast.

briansmiley committed:
Added Haggler's Charm autocast

For spell achievement farming

briansmiley committed:
Added Event logging for autocast

Console should now have a record of autocast spells

pjie commented:
Thanks!  Another thing I've noticed is that FC never goes for Easter season, presumably because the wait to get all the eggs makes it hard to value.  Perhaps it could switch to Easter specifically during cookie storms, until all the eggs are obtained?

briansmiley committed:
Tweaking Godzamok to buy back cursors instantly

Not enabling this when original AutoGodzamok is turned on, as it would make it even more insane with the constant buy/sell spam.

briansmiley committed:
Fixing autobuy/SE Autocast/Everything Must Go 

Everything Must Go was temporarily lowering the floor for bank to maintain for autocasting Spontaneous Edifice, this should now maintain the minimum balance during EMG buff

briansmiley commented:
I noticed that too. Haven't looked at anything with Season Switching yet but I'll check it out at some point.

briansmiley committed:
Added Spontaneous Edifice bank option

This should allow one to maintain a bank for future SE autocasting, while still turning on other spells if desired.

briansmiley committed:
SE Bank maintenance option

Enabled option to maintain Spontaneous edifice bank (i.e. half the cost of your most expensive building under 400) independently of AutoCast spell settings. Should allow one to maintain that bank for using SE in the future (or manually) without necessarily enabling that autocast.

briansmiley committed:
Condensed SE Bank option text

briansmiley committed:
Cleaned AutoSpell, implementing autoRigidel

Made Autospell more directly access spell casting functions rather than clicking on window elements (now that I figured out how to use minigame functions with Game.Objects[building].minigame).

Implemented autoRigidel option to automatically swap in and out rigidel when conditions are appropriate, or at least sell the right number of cursors when equipped.

briansmiley committed:
Added AutoRigidel menu option

To enable automatic swapping in and out Rigidel (+ selling cursors to get appropriate building count) for lump harvesting

briansmiley committed:
debugging autorigidel

briansmiley committed:
Fixing AutoRigidel syntax

should now properly check the current status of rigidel

briansmiley committed:
Removed debug console logs from AutoRigidel

briansmiley committed:
Fixed AutoRigidel time typo

briansmiley committed:
Fixed logging of AutoSpell

Previously logged "cast X spell" even if the conditions for casting didn't actually go off

briansmiley committed:
Fixed rigiSell and added lumpIn

Made rigiSell properly sell 0 cursors if Rigidel condition is already met, and added debugging function lumpIn to make lumps ripen in set # of mins

briansmiley committed:
Merged AutoRigidel with AutoSL menu option

briansmiley committed:
Fixing autoRigidel

Had stupid timing mistake before. Also dumped runnning AutoRigidel into AutoCookie function rather than a setInterval

briansmiley committed:
Typo on running autoRigidel

briansmiley committed:
Tweaking autoRigidel

Force game to recalculate lump time between rigiSell and clicking lump

briansmiley committed:
SL Rigidel menu hint

Added indication of AutoRigidel function

briansmiley committed:
Merge pull request #3 from Chevron42/Tinkering

Tinkering

briansmiley committed:
Added SE bank stuff

Added Spontaneous Edifice bank conditions/amounts to the general functions that determine the value of "delay", and added an exception to always pledge and buy super efficient purchases when appropriate and efficient, regardless of bank

briansmiley committed:
SE Bank fix, remove Chain Bank

Removed chain bank from consideration because they're too rare to matter. Fixed (hopefully) how Spontaneous edifice bank is compared/used.

briansmiley committed:
Update fc_main.js

briansmiley committed:
Update fc_main.js

briansmiley committed:
Merge pull request #4 from Chevron42/Tinkering

Tinkering

briansmiley committed:
Fixing case where you have no minigames yet.

briansmiley committed:
debugging

briansmiley committed:
Added exit conditions for minigame functions

briansmiley committed:
Update fc_main.js

briansmiley committed:
Commented out debugging tools

briansmiley committed:
Merge pull request #5 from Chevron42/Tinkering

Tinkering

briansmiley committed:
Adding crazy option to autogodzamok

briansmiley committed:
Implemented super crazy godzamok

briansmiley committed:
debugging

briansmiley committed:
Update fc_main.js

briansmiley committed:
Merge pull request #6 from Chevron42/Tinkering

Tinkering

briansmiley committed:
Changed wizard tower exclusion principle

Now removes Wizard towers from pool at 100 mana, not 517 wizard towers, since sugar lumps will affect mana total as well.

briansmiley committed:
tweaking autocast

Changed autocast to always go ahead and sell a CM if it doesn't have enough bank for the spell. Also made AutoCast check whether you have enough mana for a given spell (was previously just chcking if you were at max mana, which isn't enough).

briansmiley committed:
Removed mostExpensive() function

Bank for spontaneous edifice / autocast conditions now just looks at your chancemaker cost.

briansmiley committed:
Fixed SE conditions

Now properly checks if you have *half* the cost of a chancemaker, not the full cost

briansmiley committed:
Merge pull request #7 from Chevron42/Tinkering

Tinkering

briansmiley committed:
Another SE tinkering

Sets bank floor for spontaneous edifice to 0 if you don't have any CM or enough max mana to cast it.

briansmiley committed:
Adding menu option to preserve shiny wrinklers

briansmiley committed:
Maybe did the shiny wrinkler preservation change

kevmert committed:
include autobulk

include autobulk from Leonord-github

kevmert committed:
Update cc_upgrade_prerequisites.js

kevmert committed:
reincarnate timer 

increase timer for reincarnation to stop hang on reincarnation screen

kevmert committed:
Update fc_main.js

kevmert committed:
autobulk included

kevmert committed:
reincarnate delay increase

to reduce chance of hang on reincarnation screen

kevmert committed:
Update frozen_cookies.js

kevmert committed:
Update fc_userscript_loader.user.js

kevmert committed:
Update fc_bookmarklet_loader.js

kevmert committed:
Update README.md

kevmert committed:
link change

found a similar link in the readme so ill change it here too, even though it appeared to do nothing since its been this way for about 4 years

kevmert committed:
adding krazyito65 fix for incorrect refrence for userscript

briansmiley committed:
Add options and variables for Cursor Autobuy limit, not functional yet.

briansmiley committed:
Implement Cursor limit menu option

briansmiley committed:
Fix missing paren

briansmiley committed:
Fix Cursor limit function

briansmiley committed:
Merge branch 'Tinkering'

briansmiley committed:
Merge pull request #8 from kevmert/master

Adding autobulk for faster reincarnation autobuying. Hope this works.

briansmiley committed:
Add a workaround for Sugar Frenzy upgrade error

briansmiley committed:
Add missing paren on last edit

briansmiley committed:
Troubleshoot Sugar Frenzy upgrade hang bug

briansmiley committed:
Troubleshoot Sugar Frenzy bug handling

briansmiley committed:
Troubleshoot Sugar Frenzy bug handling syntax

briansmiley committed:
Add Wizard Tower buy limit toggle

briansmiley committed:
Fix Wizard Tower limit toggle typo

briansmiley committed:
Fix tower limit menu option typo

briansmiley committed:
Tidy up formatting in cc_upgrade_prerequisites.js

briansmiley committed:
Change default Cursor Limit from 0 to 500

briansmiley committed:
Fix required conditions for Wizard Tower autobuy limit

briansmiley committed:
Neaten up magic autobuy exceptions

briansmiley committed:
Remove non-functional Shiny Wrinkler menu item

briansmiley committed:
Fix wrinkler popping function loooking for shinies

briansmiley committed:
Update README.md with new features

briansmiley committed:
Remove fcBeautify function

briansmiley committed:
Fix removing fcBeautify function

briansmiley committed:
Import Beautify changes from Tinkering

briansmiley committed:
Fix FC Menu image position

briansmiley committed:
Add custom max mana for tower limit

briansmiley committed:
Add custom max mana limit menu option

briansmiley committed:
Enable setting max mana properly

briansmiley committed:
Merge pull request #9 from Chevron42/Tinkering

Tinkering

briansmiley committed:
Add note that number format toggle isn't functional

kevmert committed:
minor format fixes

also removed double of auto sugar lump harvesting

briansmiley committed:
Change AutoSwitch condition to wait for <= 1 net CPS mult

briansmiley committed:
Move FC button to the right

Fixes overlap with Santa/Dragon icons

briansmiley committed:
move button further right

joukevandermaas committed:
Blacklist milk selector, GC sound selector and background selector

briansmiley committed:
Merge pull request #11 from joukevandermaas/master

Blacklist milk selector, GC sound selector and background selector

briansmiley committed:
Merge pull request #10 from kevmert/patch-1

minor format fixes

briansmiley committed:
Add click buffs as FTHOF trigger conditions

Now FTHOF will autocast if the general CPS multiplier is high enough (like before), but also if a Dragonflight or Click Frenzy is active

Mezmerro committed:
Update fc_button.js

Mezmerro committed:
Update fc_button.js

Mezmerro committed:
Update fc_main.js

Mezmerro committed:
Update README.md

Mezmerro committed:
Update README.md

Mezmerro committed:
Update fc_userscript_loader.user.js

Mezmerro committed:
Update fc_bookmarklet_loader.js

Mezmerro committed:
Update cc_upgrade_prerequisites.js

briansmiley committed:
Merge pull request #12 from Mezmerro/master

merge Mezmerro's short numbers and button position change

briansmiley committed:
add missing 'octodecillion' number format

kevmert committed:
updating to return links (#2)

* Tweaking Godzamok to buy back cursors instantly

Not enabling this when original AutoGodzamok is turned on, as it would make it even more insane with the constant buy/sell spam.

* Fixing autobuy/SE Autocast/Everything Must Go 

Everything Must Go was temporarily lowering the floor for bank to maintain for autocasting Spontaneous Edifice, this should now maintain the minimum balance during EMG buff

* Added Spontaneous Edifice bank option

This should allow one to maintain a bank for future SE autocasting, while still turning on other spells if desired.

* SE Bank maintenance option

Enabled option to maintain Spontaneous edifice bank (i.e. half the cost of your most expensive building under 400) independently of AutoCast spell settings. Should allow one to maintain that bank for using SE in the future (or manually) without necessarily enabling that autocast.

* Condensed SE Bank option text

* Cleaned AutoSpell, implementing autoRigidel

Made Autospell more directly access spell casting functions rather than clicking on window elements (now that I figured out how to use minigame functions with Game.Objects[building].minigame).

Implemented autoRigidel option to automatically swap in and out rigidel when conditions are appropriate, or at least sell the right number of cursors when equipped.

* Added AutoRigidel menu option

To enable automatic swapping in and out Rigidel (+ selling cursors to get appropriate building count) for lump harvesting

* debugging autorigidel

* Fixing AutoRigidel syntax

should now properly check the current status of rigidel

* Removed debug console logs from AutoRigidel

* Fixed AutoRigidel time typo

* Fixed logging of AutoSpell

Previously logged "cast X spell" even if the conditions for casting didn't actually go off

* Fixed rigiSell and added lumpIn

Made rigiSell properly sell 0 cursors if Rigidel condition is already met, and added debugging function lumpIn to make lumps ripen in set # of mins

* Merged AutoRigidel with AutoSL menu option

* Fixing autoRigidel

Had stupid timing mistake before. Also dumped runnning AutoRigidel into AutoCookie function rather than a setInterval

* Typo on running autoRigidel

* Tweaking autoRigidel

Force game to recalculate lump time between rigiSell and clicking lump

* SL Rigidel menu hint

Added indication of AutoRigidel function

* Added SE bank stuff

Added Spontaneous Edifice bank conditions/amounts to the general functions that determine the value of "delay", and added an exception to always pledge and buy super efficient purchases when appropriate and efficient, regardless of bank

* SE Bank fix, remove Chain Bank

Removed chain bank from consideration because they're too rare to matter. Fixed (hopefully) how Spontaneous edifice bank is compared/used.

* Update fc_main.js

* Update fc_main.js

* Fixing case where you have no minigames yet.

* debugging

* Added exit conditions for minigame functions

* Update fc_main.js

* Commented out debugging tools

* Adding crazy option to autogodzamok

* Implemented super crazy godzamok

* debugging

* Update fc_main.js

* Changed wizard tower exclusion principle

Now removes Wizard towers from pool at 100 mana, not 517 wizard towers, since sugar lumps will affect mana total as well.

* tweaking autocast

Changed autocast to always go ahead and sell a CM if it doesn't have enough bank for the spell. Also made AutoCast check whether you have enough mana for a given spell (was previously just chcking if you were at max mana, which isn't enough).

* Removed mostExpensive() function

Bank for spontaneous edifice / autocast conditions now just looks at your chancemaker cost.

* Fixed SE conditions

Now properly checks if you have *half* the cost of a chancemaker, not the full cost

* Another SE tinkering

Sets bank floor for spontaneous edifice to 0 if you don't have any CM or enough max mana to cast it.

* Adding menu option to preserve shiny wrinklers

* Maybe did the shiny wrinkler preservation change

* Add options and variables for Cursor Autobuy limit, not functional yet.

* Implement Cursor limit menu option

* Fix missing paren

* Fix Cursor limit function

* Add a workaround for Sugar Frenzy upgrade error

* Add missing paren on last edit

* Troubleshoot Sugar Frenzy upgrade hang bug

* Troubleshoot Sugar Frenzy bug handling

* Troubleshoot Sugar Frenzy bug handling syntax

* Add Wizard Tower buy limit toggle

* Fix Wizard Tower limit toggle typo

* Fix tower limit menu option typo

* Tidy up formatting in cc_upgrade_prerequisites.js

* Change default Cursor Limit from 0 to 500

* Fix required conditions for Wizard Tower autobuy limit

* Neaten up magic autobuy exceptions

* Remove non-functional Shiny Wrinkler menu item

* Fix wrinkler popping function loooking for shinies

* Update README.md with new features

* Remove fcBeautify function

* Fix removing fcBeautify function

* Fix FC Menu image position

* Add custom max mana for tower limit

* Add custom max mana limit menu option

* Enable setting max mana properly

* Add note that number format toggle isn't functional

* minor format fixes

also removed double of auto sugar lump harvesting

* Change AutoSwitch condition to wait for <= 1 net CPS mult

* Move FC button to the right

Fixes overlap with Santa/Dragon icons

* move button further right

* Blacklist milk selector, GC sound selector and background selector

* Add click buffs as FTHOF trigger conditions

Now FTHOF will autocast if the general CPS multiplier is high enough (like before), but also if a Dragonflight or Click Frenzy is active

* Update fc_button.js

* Update fc_button.js

* Update fc_main.js

* Update README.md

* Update README.md

* Update fc_userscript_loader.user.js

* Update fc_bookmarklet_loader.js

* Update cc_upgrade_prerequisites.js

* add missing 'octodecillion' number format

kevmert committed:
update userscript back to correct link location

kevmert committed:
update userscript back to correct link location

kevmert committed:
update userscript back to correct link location

pachs committed:
Attempt to get Achievements to show when earned

pachs committed:
Fix achievements not showing up

briansmiley committed:
Merge pull request #13 from kevmert/master

merging the updates

briansmiley committed:
Merge pull request #14 from pachs/chevron-with-achievements-fix

Show Achievements fix
